File: Miguel Franco, Oswaldo Romero/Tareas/Identificar Variables/Variable.py
# Introducir un dato
dato = input("Introduce un dato: ")

# Determinar el tipo de dato ingresado
if dato.lower() == "true" or dato.lower() == "false":
    tipo_dato = "bool"
elif len(dato) == 1:
    tipo_dato = "char"
elif "." in dato:
    try:
        float(dato)
        tipo_dato = "float"
    except ValueError:
        tipo_dato = "string"
else:
    try:
        int(dato)
        tipo_dato = "int"
    except ValueError:
        tipo_dato = "string"

# Mostrar el tipo de dato
print(f"El tipo de dato ingresado es: {tipo_dato}")

File: Miguel Franco, Oswaldo Romero/Tareas/Identificar Variables/variable.php
<!DOCTYPE html>
<html lang="es">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Determinar Tipo de Dato</title>
</head>
<body>
    <h1>Determinar Tipo de Dato</h1>
    <form action="" method="post">
        <label for="txtDato">Ingrese un dato:</label><br>
        <input type="text" id="txtDato" name="txtDato"><br><br>
        <input type="submit" value="Determinar Tipo">
    </form>

    <?php
    if ($_SERVER['REQUEST_METHOD'] === 'POST') {
        $dato = $_POST['txtDato'];

        if ($dato === 'true' || $dato === 'false') {
            $tipo_dato = 'bool';
        } elseif (strlen($dato) === 1) {
            $tipo_dato = 'char';
        } elseif (is_numeric($dato)) {
            if (strpos($dato, '.') !== false) {
                $tipo_dato = 'float';
            } else {
                $tipo_dato = 'int';
            }
        } else {
            $tipo_dato = 'string';
        }

        echo "<p>El tipo de dato ingresado es: " . $tipo_dato . "</p>";
    }
    ?>
</body>
</html>
